conformance/lang/expressions/multiplicative-expr/division.balt

```diff
@@ -4033,3 +4033,922 @@ Description: Test integer division panic at module-level.
+    io:println(v); // @output 1.25
+
+    v = q / j;
+    io:println(v); // @output 0.625
+
+    v = q / k;
+    io:println(v); // @output 0.625
+
+    v = q / m;
+    io:println(v); // @output 0.5
+
+    v = q / n;
+    io:println(v); // @output 0.25
+
+    v = q / p;
+    io:println(v); // @output 0.625
+}
+
+Test-Case: output
+Description: Test optional float division with optional and non-optional subtypes of integers initialized with nil-literal.
+Labels: DecimalNumber, float, int, multiplicative-expr, nil-literal, numeric-nil-lifting
+
+function init() {
+    int:Signed8? a = ();
+    int:Signed16? b = ();
+    int:Signed32? c = ();
+    int:Unsigned8? d = ();
+    int:Unsigned16? e = ();
+    int:Unsigned32? f = ();
+    byte? g = ();
+
+    float? h = 2.5;
+
+    float? v = h / a;
+    io:println(v); // @output ()
+
+    v = h / b;
+    io:println(v); // @output ()
+
+    v = h / c;
+    io:println(v); // @output ()
+
+    v = h / d;
+    io:println(v); // @output ()
+
+    v = h / e;
+    io:println(v); // @output ()
+
+    v = h / f;
+    io:println(v); // @output ()
+
+    v = h / g;
+    io:println(v); // @output ()
+}
+
+Test-Case: output
+Description: Test type of float division with integers by infering.
+Labels: DecimalNumber, float, int, multiplicative-expr, numeric-nil-lifting
+
+const int constInt = 5;
+const float constFloat = 2.5;
+
+function init() {
+    float a = 2.5;
+    int b = 5;
+
+    var c = a / b;
+    io:println(c is float); // @output true
+
+    var e = a / constInt;
+    io:println(e is float); // @output true
+
+    var g = constFloat / b;
+    io:println(g is float); // @output true
+
+    var i = constFloat / constInt;
+    io:println(i is float); // @output true
+}
+
+Test-Case: output
+Description: Test type of optional float division with optional and non-optional integers by infering.
+Labels: DecimalNumber, float, int, module-const-decl, multiplicative-expr, numeric-nil-lifting
+
+const int constInt = 5;
+const float constFloat = 2.5;
+
+function init() {
+    float? a = 2.5;
+    int? b = 5;
+
+    var c = a / b;
+    io:println(c is float); // @output true
+
+    var e = a / constInt;
+    io:println(e is float); // @output true
+
+    var g = constFloat / b;
+    io:println(g is float); // @output true
+
+    var i = constFloat / constInt;
+    io:println(i is float); // @output true
+}
+
+Test-Case: output
+Description: Test type of float division with integers results Infinity and NaN.
+Labels: DecimalNumber, float, float:Infinity, float:NaN, int, multiplicative-expr
+
+function init() {
+    float a = 8388608333e+298;
+    int b = 20;
+    float c = float:Infinity;
+    float d = float:NaN;
+    int e = 0;
+
+    float v = c / b;
+    io:println(v); // @output Infinity
+
+    v = d / b;
+    io:println(v); // @output NaN
+
+    v = a / e;
+    io:println(v); // @output Infinity
+
+    v = c / e;
+    io:println(v); // @output Infinity
+
+    v = d / e;
+    io:println(v); // @output NaN
+}
+
```

Comment:
Let's also add some tests with different user-defined subtypes of int and float
Ex:
```ballerina
const int A = 1;
const int B = 2;
type IntType1 int:Signed8|int:Unsigned8;
type IntType2 A|B;
type IntType3 A|int:Signed16;
type IntType4 IntType2|IntType3;
```